### PR TITLE
Short-circuit efficient latex update if old and new values are identical

### PR DESCRIPTION
**Pull request by sclower**

### CODE REVIEW

File: src/services/latex.js [hunk 1]
```diff
@@ -118,7 +118,10 @@ Controller.open(function(_, super_) {
...
   };
   _.renderLatexMathEfficiently = function (latex) {
-    var oldLatex, oldClassification;
+    var oldLatex = this.exportLatex();
+    if (oldLatex === latex) return true;
+
+    var oldClassification;
     var classification = this.classifyLatexForEfficientUpdate(latex);
     if (classification) {
       oldLatex = this.exportLatex();
```
Comment (mikehaverstock) [inline src/services/latex.js:127]: we probably don't need this second call to "exportLatex()"

File: src/services/latex.js [hunk 2]
```diff
@@ -268,6 +270,7 @@ Controller.open(function(_, super_) {
     return true;
   };
   _.renderLatexMathFromScratch = function (latex) {
+    this.rendered = true;
```
Comment (jwmerrill) [inline src/services/latex.js:273]: What does this do? Wondering if this is functional, or left over from debugging.

File: src/services/latex.js [hunk 3]
```diff
@@ -118,10 +118,14 @@ Controller.open(function(_, super_) {
     }
   };
   _.renderLatexMathEfficiently = function (latex) {
-    var oldLatex, oldClassification;
+    var root = this.root;
+    var oldLatex = this.exportLatex();
+    if (root.ends[L] && root.ends[R] && oldLatex === latex) {
```
Comment (jwmerrill) [inline src/services/latex.js:123]: Can you explain a little more about why this is necessary?

What's a situation where `exportLatex` returns something, but these aren't populated? Are we deferring rendering if the mathquill isn't visible or something like that?

Comment (sclower) [reply src/services/latex.js:123]: No, but it's possible for the render method to be called more than once via different paths, in particular if latex is being set via a math command directly. Without checking for MQ nodes, we could get into a state where the latex was set but we never created DOM for it.

**Review by jwmerrill — APPROVED** [review]

**Pull request merged**